src/Features/Core/Portable/Workspace/CompileTimeSolutionProvider.cs

```diff
@@ -67,7 +67,12 @@ public Solution GetCompileTimeSolution(Solution designTimeSolution)
                 if (designTimeSolution.WorkspaceVersion == _correspondingDesignTimeSolutionVersion)
                 {
                     Contract.ThrowIfNull(_lazyCompileTimeSolution);
-                    return _lazyCompileTimeSolution;
+
+                    // Only re-use when the lazy solution is from the same branch as the requested solution.
+                    if (_lazyCompileTimeSolution.BranchId == designTimeSolution.BranchId)
```

Comment:
>_lazyCompileTimeSolution [](http://example.com/codeflow?start=24&length=24)

Should we cache 2 snapshots (branches)?

Reply:
yeah that seems reasonable, currently it ends up flip-flopping depending on how the requests come in.  I'll have one cached snapshot for the primary branch and another item cached for a different branch (note that the branchId for the LSP solution changes on every text change since we fork from the workspace each time)

Reply:
@tmat updated to include a cache for the primary branch as well as another snapshot cache for forked requests